src/transformers/models/donut/processing_donut.py

```diff
@@ -156,14 +156,17 @@ def token2json(self, tokens, is_inner_value=False, added_vocab=None):
         output = {}
 
         while tokens:
-            start_token = re.search(r"<s_(.*?)>", tokens, re.IGNORECASE)
-            if start_token is None:
+            potential_start = re.search(r"<s_", tokens, re.IGNORECASE)
+            if potential_start is None:
                 break
-            key = start_token.group(1)
+            potential_end = re.search(r">", tokens[potential_start.end() :], re.IGNORECASE)
+            if potential_end is None:
+                break
+            start_token = tokens[potential_start.start() : potential_start.end() + potential_end.end()]
```

Comment:
nit: I would add a comment (or create intermediary vars) to help explaining the why behind `potential_start.end() + potential_end.end()` (`potential_end` searches over a substring)

Reply:
Yes, good point! I rewrote that bit.